### PR TITLE
Add compile time version information

### PR DESCRIPTION
**Pull request by fxha**
| Q                | A
| ---------------- | ---
| License          | MIT

### Description

I added the version information to the main and renderer process at compile time to simplify access and also outsourced the environment definitions.

### CODE REVIEW

File: .electron-vue/marktextEnvironment.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const GitRevisionPlugin = require('git-revision-webpack-plugin')
+const { version } = require('../package.json')
+
+const getEnvironmentDefinitions = function () {
+  const gitRevisionPlugin = new GitRevisionPlugin()
+  const isOfficialRelease = !!process.env.MARKTEXT_IS_OFFICIAL_RELEASE
```
Comment (Jocs) [inline .electron-vue/marktextEnvironment.js]: I forget how to set the `MARKTEXT_IS_OFFICIAL_RELEASE ` variable. where to set it?

File: src/main/cli.js [hunk 2]
```diff
@@ -12,8 +11,7 @@ for (const arg of process.argv) {
     global.MARKTEXT_SAFE_MODE = true
     continue
   } else if (arg === '--version') {
-    const gitInfo = global.MARKTEXT_IS_OFFICIAL_RELEASE ? '' : `(${global.MARKTEXT_GIT_INFO})`
-    console.log(`Mark Text: ${app.getVersion()} ${gitInfo}
+    console.log(`Mark Text: ${global.MARKTEXT_VERSION_STRING}
```
Comment (Jocs) [inline src/main/cli.js]: @fxha How to set `MARKTEXT_VERSION_STRING `?

File: .electron-vue/marktextEnvironment.js (hunk 1 above)
Comment (fxha) [reply .electron-vue/marktextEnvironment.js]: You have to set the environment variable `MARKTEXT_IS_OFFICIAL_RELEASE` to not null (e.g. `export MARKTEXT_IS_OFFICIAL_RELEASE=1`). It's used to not add the commit hash to the version string.

File: src/main/cli.js (hunk 2 above)
Comment (fxha) [reply src/main/cli.js]: It's automatically set in line 15/16 and is read from the `package.json` `version` field.

**Pull request merged**